File: 1.Python/Interview/4.Character_Occurance.py
"""
Write a program to count the occurrences of each character in a string.
"""
from collections import Counter
def char_counter(text):
    return Counter(text)

print(char_counter("Hello"))

chars = Counter("Ramanan")

"""
Write a program to count the occurrences of each character in a string.
#without inbuilt function
"""

def character_count(text):
    d = {}
    for c in text:
        if c not in d:
            d[c] = 1
        else:
            d[c] += 1
    
    return d

print(character_count("Hello"))

for k,v in character_count("Hello").items():
    print(f"{k} : {v}")

File: 3.JavaScript/Interview/4.Character_Occurance.js
/*
Write a program to count the occurrences of each character in a string.
*/

const text = "Radar"

let count = {}

for (let c of text.toLocaleLowerCase())
{
  if (!count[c])
    {
        count[c] = 1
    }
    else{
        count[c] += 1
    }  
}

console.log(count)

for (let [k,v] of Object.entries(count))
{
    console.log(k , v)
}